### PR TITLE
EVG-15638 Implement History Table Test Search component

### PR DESCRIPTION
**Pull request by minnakt**
[EVG-15638](https://jira.mongodb.org/browse/EVG-15638)

### Description 
This PR adds the `TestSearch` component to the `VariantHistory.tsx` and `TaskHistory.tsx` pages. Note that the component is not fully functional at the moment (needs to be wired up to the table in another ticket).

Note: also encompasses [EVG-15675](https://jira.mongodb.org/browse/EVG-15675).

### Screenshots
https://user-images.githubusercontent.com/47064971/138810406-d9a05842-f605-41ca-b94f-b92d369a4d15.mov

### Testing 
* Added a unit test file `TestSearch.test.tsx`
* Added to storybook `TestSearch.stories.tsx`
* Modified tests related to `FilterBadges` to accommodate for new props

### CODE REVIEW

File: src/components/HistoryTable/HistoryTableTestSearch/HistoryTableTestSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+`;
+
+const TextInputWrapper = styled.div`
+  position: relative;
+`;
+
+const StyledIcon = styled(Icon)`
+  position: absolute;
+  bottom: 10px;
+  right: 10px;
```
Comment (khelif96) [inline src/components/HistoryTable/HistoryTableTestSearch/HistoryTableTestSearch.tsx:120]: nit. you can also use flex box to center this icon in the text input with out relying on pixel values.

**Review by khelif96 — COMMENTED** [review]
Last few comments.

File: src/components/FilterBadges/index.tsx [hunk 2]
```diff
@@ -2,30 +2,25 @@ import styled from "@emotion/styled";
...
 const { parseQueryString } = queryString;
 
-const hiddenQueryParams = new Set([
-  ChartToggleQueryParams.chartType,
-  PatchTasksQueryParams.Statuses,
-  MainlineCommitQueryParams.SkipOrderNumber,
-]);
-export const FilterBadges: React.FC = () => {
+interface FilterBadgesProps {
+  queryParamsToIgnore: Set<string>;
```
[the remark this answers is not in the document]
Comment (minnakt) [reply src/components/FilterBadges/index.tsx:13]: makes a lot of sense 🤔 will change!

File: src/components/FilterBadges/FilterBadges.stories.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ export const Default = () => {
     updateQueryParams({ [badgeKey]: params });
   };
   button("Add Badge", addBadge);
-  return <FilterBadges queryParamsToIgnore={new Set([])} />;
+  return <FilterBadges queryParamsToDisplay={new Set(["buildvariants"])} />;
```
Comment (khelif96) [inline src/components/FilterBadges/FilterBadges.stories.tsx:35]: nit. you can use one of the enums from `types/commits`

**Review by khelif96 — APPROVED** [review]

**Pull request merged**